CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+cmake_minimum_required(VERSION 3.2)
+project(OpenDIS)
+
+## Libraries
+
+# Add src/ to the include directories
+include_directories(src)
+
+# create list of DIS6 source files
+file(GLOB DIS6_SOURCES
+  "src/dis6/*.cpp"
+  "src/utils/*.cpp"
+)
+# Define ExampleSender Executable
+add_library(OpenDIS6 SHARED ${DIS6_SOURCES})
+# Add compile definition EXPORT_LIBRARY for DIS6 msLibMacro.h
+target_compile_definitions(OpenDIS6 PRIVATE EXPORT_LIBRARY)
```

Comment:
Would be nice to have `target_include_directories` and exported targets such that this can be used with `find_package`.

Reply:
like so? 
```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,8 +3,6 @@ project(OpenDIS)
 
 ## Libraries
 
-# Add src/ to the include directories
-include_directories(src)
 
 # create list of DIS6 source files
 file(GLOB DIS6_SOURCES
@@ -13,6 +11,8 @@ file(GLOB DIS6_SOURCES
 )
 # Define ExampleSender Executable
 add_library(OpenDIS6 SHARED ${DIS6_SOURCES})
+# Add src/ to the include directories to OpenDIS6
+target_include_directories(OpenDIS6 PUBLIC src)
 # Add compile definition EXPORT_LIBRARY for DIS6 msLibMacro.h
 target_compile_definitions(OpenDIS6 PRIVATE EXPORT_LIBRARY)
 
@@ -23,6 +23,8 @@ file(GLOB DIS7_SOURCES
 )
 # Define ExampleSender Executable
 add_library(OpenDIS7 SHARED ${DIS7_SOURCES})
+# Add src/ to the include directories to OpenDIS6
+target_include_directories(OpenDIS7 PUBLIC src)
 # Add compile definition EXPORT_LIBRARY for DIS7 msLibMacro.h
 target_compile_definitions(OpenDIS7 PRIVATE EXPORT_LIBRARY)
 
@@ -49,6 +51,8 @@ add_executable(ExampleSender ${EX_SENDER_SOURCES})
 # Link OpenDIS into ExampleSender
 target_link_libraries(ExampleSender PRIVATE OpenDIS6)
 target_link_libraries(ExampleSender PRIVATE SDL2main SDL2 SDL2_net)
+# add src to ExampleSender include directories
+target_include_directories(ExampleSender PRIVATE src)
 
 # create list of ExampleReceiver source files
 file(GLOB EX_RECEIVER_SOURCES
@@ -59,11 +63,13 @@ file(GLOB EX_RECEIVER_SOURCES
   "examples/EntityStatePduProcessor.cpp"
 )
 
-# Define ExampleReciever Executable
+# Define ExampleReceiver Executable
 add_executable(ExampleReceiver ${EX_RECEIVER_SOURCES})
 # Link OpenDIS into ExampleReceiver
 target_link_libraries(ExampleReceiver PRIVATE OpenDIS6)
 target_link_libraries(ExampleReceiver PRIVATE SDL2main SDL2 SDL2_net)
+# add src to ExampleSender include directories
+target_include_directories(ExampleReceiver PRIVATE src)
```

Reply:
You can check #33 for reference. The solution implemented there had these capabilities.